### PR TITLE
Support Hostname property in VM creation

### PR DESCRIPTION
**Pull request by mattcui**
This PR is for addressing [tracker 139523259](https://www.pivotaltracker.com/story/show/139523259) and [issue 211](https://github.com/cloudfoundry/bosh-softlayer-cpi-release/issues/211)

### CODE REVIEW

**Comment by mattcui** [comment]
@maximilien it's ready for review, thanks.

File: src/bosh-softlayer-cpi/action/create_vm.go [hunk 1]
```diff
@@ -77,10 +77,18 @@ func (a CreateVMAction) Run(agentID string, stemcellCID StemcellCID, cloudProps
 func (a CreateVMAction) updateCloudProperties(cloudProps *VMCloudProperties) {
 	a.vmCloudProperties = cloudProps
 
-	if len(cloudProps.BoshIp) == 0 || cloudProps.Baremetal {
-		a.vmCloudProperties.VmNamePrefix = cloudProps.VmNamePrefix
+	if cloudProps.NotDeployedByDirector {
```
Comment (maximilien) [inline src/bosh-softlayer-cpi/action/create_vm.go]: I dislike the negative name here. So for me, something like: `if !cloudProps.DeployedByDirector { ...` reads better. But perhaps that's just me.

File: src/bosh-softlayer-cpi/action/create_vm.go [hunk 2]
```diff
@@ -77,10 +77,18 @@ func (a CreateVMAction) Run(agentID string, stemcellCID StemcellCID, cloudProps
...
+		if len(cloudProps.Hostname) == 0 {
+			a.vmCloudProperties.VmNamePrefix = cloudProps.VmNamePrefix
+		} else {
+			a.vmCloudProperties.VmNamePrefix = cloudProps.Hostname
+		}
 	} else {
-		a.vmCloudProperties.VmNamePrefix = cloudProps.VmNamePrefix + TimeStampForTime(time.Now().UTC())
+		if len(cloudProps.Hostname) == 0 {
```
Comment (maximilien) [inline src/bosh-softlayer-cpi/action/create_vm.go]: I feel like lines 81-85 should be in private method with a param of optional timestamp... This avoids duplicating that code and make the if/else/if/else logic a bit better.

File: src/bosh-softlayer-cpi/bin/test-unit [hunk 3]
```diff
@@ -3,6 +3,10 @@
 (
   set -e
```
Comment (maximilien) [inline src/bosh-softlayer-cpi/bin/test-unit]: Why is there two sets of changes to this file?

File: src/bosh-softlayer-cpi/action/create_vm.go (hunk 2 above)
Comment (mattcui) [inline src/bosh-softlayer-cpi/action/create_vm.go]: Changed the property name to `DeployedByBoshCLI`

File: src/bosh-softlayer-cpi/bin/test-unit (hunk 3 above)
Comment (mattcui) [inline src/bosh-softlayer-cpi/bin/test-unit]: test-unit didn't work after Grace combined source into release projects, I corrected the file.

File: src/bosh-softlayer-cpi/action/create_vm.go (hunk 1 above)
Comment (mattcui) [inline src/bosh-softlayer-cpi/action/create_vm.go]: Changed the property name to `DeployedByBoshCLI`

File: src/bosh-softlayer-cpi/action/create_vm.go (hunk 2 above)
Comment (mattcui) [inline src/bosh-softlayer-cpi/action/create_vm.go]: Addressed your comment, add a private function `updateHostNameInCloudProps`

**Comment by mattcui** [comment]
@maximilien I already addressed all of your comments, please help review again. Thanks.

**Comment by maximilien** [comment]
OK merging.

**Pull request merged**